pkg/karmadactl/promote.go

```diff
@@ -229,9 +245,67 @@ func RunPromote(karmadaConfig KarmadaConfig, opts CommandPromoteOption, args []s
 		return fmt.Errorf("failed to get gvr from %q: %v", opts.gvk, err)
 	}
 
+	if opts.Deps {
+		err := promoteDeps(obj, opts, mapper, gvr, controlPlaneRestConfig)
+		if err != nil {
+			return fmt.Errorf("failed to promote relevant resources of gvr %q automatically: %v", opts.gvk, err)
+		}
+	}
 	return promote(controlPlaneRestConfig, obj, gvr, opts)
 }
 
+func promoteDeps(obj *unstructured.Unstructured, opts CommandPromoteOption, mapper meta.RESTMapper, gvr schema.GroupVersionResource, controlPlaneRestConfig *rest.Config) error {
+	interpreter := defaultinterpreter.NewDefaultInterpreter()
```

Comment:
Is it limited if we only use `defaultinterpreter`?

Reply:
What do you think ? should we support customizedInterpreter in promote?

Reply:
> I think it is necessary because we can promote any resource, especially support CRD/CR.

Agree,But it looks a little difficult. As webhook handle resource in control plane,but `promote `handles resources in member cluster.
Let me doing some research on it.

Reply:
> > I think it is necessary because we can promote any resource, especially support CRD/CR.
> 
> Agree,But it looks a little difficult. As webhook handles resource in control plane,but `promote `handles resources in member cluster. Let me doing some research on it.


Hey @duanmengkk I was looking into the issue could you give a little bit more context on the above comment you made. Thanks already